File: Archi/Archi/ClientApp/src/model/Book.ts
import {Genre} from "../enum/Genre";

export class Book {
    public id: string;
    public name: string;
    public genre: Genre;
    public author: string;
    public isRemoved: boolean;
    
    constructor(id: string, name: string, genre: Genre, author: string, isRemoved: boolean) {
        this.id = id;
        this.name = name;
        this.genre = genre;
        this.author = author;
        this.isRemoved = isRemoved;
    }
}

File: Archi/Archi/Domain/Model/Book.cs
using System;
using System.Net;
using Archi.Infrostructure;

namespace Archi.Domain.Model
{
    public class Book
    {
        public Guid Id { get; set; }
        public String Name { get; set; }
        public GenreBook Genre { get; set; }
        public String Author { get; set; }
        public Boolean IsRemoved { get; set; }

        public Book() { }

        public Book(Guid id, String name, GenreBook genre, String author, Boolean isRemoved)
        {
            Id = id;
            Name = name;
            Genre = genre;
            Author = author;
            IsRemoved = isRemoved;
        }
    }
}